### ISSUE TITLE
intermittent twisted.test.test_ssl.SpammyTLSTestCase.testUnTLS failure on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2918]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2918|
|Type|defect|
|Created|2007-11-20 20:53:00Z|

```
===============================================================================
[ERROR]: twisted.test.test_ssl.SpammyTLSTestCase.testUnTLS

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<<class 'twisted.internet.tcp.TLSConnection'> to ('127.0.0.1', 64745) at 17bd4d0>
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__2918 2918
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1195591980000000 1195591980000000
changetime__1211900230000000 1211900230000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

Related, `twisted.test.test_ssl.TLSTestCase.testUnTLS` can fail in the same way on the same platform.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

(In [23717]) Merge test-ssl-iterate-2374

Author: therve
Reviewer: exarkun
Fixes [#2374](https://github.com/twisted/twisted/issues/2374)
Fixes [#2918](https://github.com/twisted/twisted/issues/2918)

Remove reactor.iterate calls from test_ssl.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
